src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java

```diff
@@ -1059,6 +1059,14 @@ public Role createRole(GuildImpl guild, DataObject roleJson, long guildId)
             .setHoisted(roleJson.getBoolean("hoist"))
             .setColor(color == 0 ? Role.DEFAULT_COLOR_RAW : color)
             .setMentionable(roleJson.getBoolean("mentionable"));
+        if(roleJson.hasKey("tags"))
+        {
+            role.setType(Role.RoleType.fromTags(roleJson.getObject("tags")));
+        }
+        else
+        {
+            role.setType(Role.RoleType.NORMAL);
+        }
```

Comment:
This is not a good way to handle tags. You should be storing the tags in a map so that you can also retrieve the associated values. For example, which bot a role belongs to.
```suggestion
        if (roleJson.hasKey("tags"))
            role.setType(Role.RoleType.fromTags(roleJson.getObject("tags")));
        else
            role.setType(Role.RoleType.NORMAL);
```